infrahub_sdk/spec/processors/data_processor.py

```diff
@@ -0,0 +1,10 @@
+from abc import ABC, abstractmethod
+from typing import Any
+
+
+class DataProcessor(ABC):
+    """Abstract base class for data processing strategies"""
+
+    @abstractmethod
+    async def process_data(self, data: list[dict[str, Any]]) -> list[dict[str, Any]]:
```

Comment:
Realising that when looking at the we're just sending opaque data back and forth without really knowing what to expect. While not tied to this PR it makes me wonder about the object specs that we have. We should be able to read the yaml files and convert them into a known structure instead, but that would be some follow up work.